packages/amplify-graphql-transformer-core/src/migration/migrators/auth/defaultAuth.ts

```diff
@@ -0,0 +1,56 @@
+import { getAuthRules, hasAuthDirectives, addAuthRuleToNode } from '.'
+import { createAuthRule } from '../generators'
+
+const defaultAuthModeMap: Map<string, string> = new Map<string, string>([
+  ['apiKey', 'public'],
+  ['iam', 'private'],
+  ['userPools', 'private'],
+  ['oidc', 'private']
+]);
+
+function getDefaultAuthRule(authMode: any, rules: any) {
+  const parsedRules = rules.map((rule: any) => ({
+    provider: rule.fields.find((r: any) => r.name.value === 'provider'),
+    strategy: rule.fields.find((r: any) => r.name.value === 'allow')
+  })).map((r: any) => ({
+    strategy: r.strategy.value.value,
+    provider: r.provider?.value?.value
+  }));
+
+  const foundRules = parsedRules.filter((r: any) => r.strategy === defaultAuthModeMap.get(authMode));
+  if (foundRules.length === 0) {
+    return null;
+  }
+
+  if (authMode === "iam" && foundRules.filter((r: any) => r.provider === 'iam').length !== 0) {
+    return foundRules.find((r: any) => r.provider === 'iam');
```

Comment:
The `filter()` operation here seems unnecessary/inefficient since you're calling `find()` on the next line with th same condition. You could just store the result of `find()` in a variable.